compact.go

```diff
@@ -420,12 +420,14 @@ func (c *LeveledCompactor) Compact(dest string, dirs []string, open []*Block) (u
 		if meta.Stats.NumSamples == 0 {
 			for _, b := range bs {
 				b.meta.Compaction.Deletable = true
-				if err = writeMetaFile(c.logger, b.dir, &b.meta); err != nil {
+				n, err := writeMetaFile(c.logger, b.dir, &b.meta)
+				if err != nil {
 					level.Error(c.logger).Log(
 						"msg", "Failed to write 'Deletable' to meta file after compaction",
 						"ulid", b.meta.ULID,
 					)
 				}
+				b.numBytesMeta = n
```

Comment:
Why we set numBytes to new block, but we ignore the other sizes later on?

Reply:
actually this is setting the numBytesMeta for the existing blocks. Existing blocks are already open so need to set the size information when it changes.

Later on it is creating new blocks and these would have their size calculated when they are open at 'db.reload()'.

Reply:
Good point, LGTM.